File: objc/test/Ice/adapterDeactivation/Collocated.m
// **********************************************************************
//
// Copyright (c) 2003-2008 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICETOUCH_LICENSE file included in this distribution.
//
// **********************************************************************

#import <Ice/Application.h>
#import <ServantLocatorI.h>
#import <TestCommon.h>
#import <Test.h>


@interface TestServer : Application
{
public:

-(int) run:(int)XXX XXX:(char*[])XXX;
};

int
main(int argc, char* argv[])
{
    TestServer app;
    return app.main(argc, argv);
}

int
-(XXX) run:(int argc, char* argv[])
{
    communicator()->getProperties()->setProperty("TestAdapter.Endpoints" :@"default -p 12010 -t 10000");
    id<ICEObjectAdapter> adapter = communicator()->createObjectAdapter("TestAdapter");
    id<ServantLocator> locator = [[ServantLocatorI alloc] init];
    [adapter addServantLocator:locator XXX:@""];

-(TestIntfPrx) allTests:(id<Communicator>)XXX;
    allTests(communicator());

    [adapter waitForDeactivate];
    return EXIT_SUCCESS;
}

File: cs/test/Ice/adapterDeactivation/Collocated.cs
// **********************************************************************
//
// Copyright (c) 2003-2009 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

using Test;
using System;
using System.Diagnostics;
using System.Reflection;

[assembly: CLSCompliant(true)]

[assembly: AssemblyTitle("IceTest")]
[assembly: AssemblyDescription("Ice test")]
[assembly: AssemblyCompany("ZeroC, Inc.")]

public class Collocated
{
    internal class App : Ice.Application
    {
        public override int run(string[] args)
        {
            communicator().getProperties().setProperty("TestAdapter.Endpoints", "default -p 12010 -t 10000");
            Ice.ObjectAdapter adapter = communicator().createObjectAdapter("TestAdapter");
            Ice.ServantLocator locator = new ServantLocatorI();
            adapter.addServantLocator(locator, "");

            AllTests.allTests(communicator());
            
            adapter.waitForDeactivate();
            return 0;
        }
    }
    
    public static void Main(string[] args)
    {
        Debug.Listeners.Add(new ConsoleTraceListener());

        App app = new App();
        int result = app.main(args);
        if(result != 0)
        {
            System.Environment.Exit(result);
        }
    }
}
